worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/PaperweightGetBlocks.java

```diff
@@ -894,9 +881,9 @@ public char[] update(int layer, char[] data, boolean aggressive) {
                 } else {
                     // The section's palette is the global block palette.
                     for (int i = 0; i < 4096; i++) {
-                        char paletteVal = data[i];
+                        char paletteVal = (char) data.getAt(i);
```

Comment:
I think this (and the equivalent in all other adapters) should use int rather than char, else, the method breaks down when using an IntDataArray. Methods that return char should also be changed to int in adapters (e.g. `adaptToChar`, `adaptToChar` etc. The cached char[] array for `ibdToStateOrdinal` can also just be int[] - the memory impact is pretty negligible (in the 100s of KB)